internal/manifests/collector/deployment.go

```diff
@@ -35,7 +35,7 @@ func Deployment(params manifests.Params) (*appsv1.Deployment, error) {
 			Annotations: annotations,
 		},
 		Spec: appsv1.DeploymentSpec{
-			Replicas: manifestutils.GetInitialReplicas(params.OtelCol),
+			Replicas: manifestutils.GetDesiredReplicas(params.OtelCol),
```

Comment:
I believe if `GetDesiredReplicas` will produce the same effect as described on #2585

Reply:
I also tested this manually and have noticed no problems. We should add a e2e tests for this scenario in a follow-up, but I'm not going to block this PR with that requirement. I'm going to merge now. @jaronoff97 can you make sure to pick it up for 0.138.0?